pkg/oauth/external/handler_test.go

```diff
@@ -149,3 +264,74 @@ func TestRedirectingStateError(t *testing.T) {
 		t.Errorf("Expected original error back, got %#v", err)
 	}
 }
+
+type attributes struct {
+	authorizer.Attributes
+}
+
+func (attributes) GetUser() user.Info      { return nil }
+func (attributes) GetVerb() string         { return "" }
+func (attributes) IsResourceRequest() bool { return false }
+
+type provider struct {
```

Comment:
@s-urbaniak showed me this neat trick: https://github.com/openshift/oauth-server/pull/107/files#diff-df8eb2f6bcbdb1bd156b4feaf6a76f7879dbeae80a28ce00add3be51943a4185R284-R290

It reliefs you of the necessity to define function you won't use, just to satisfy an interface.

Reply:
done